IceCubeOpenSource/pisa
# Cleanup numba_osc code

* add unit test for untested function
* add wrappers so all tests can be run whether TARGET={cpu,cuda}
* define reference test values to test output against (some based on current outputs). while not perfect (maybe we have a bug now?) at least we can test that code doesn't change from now
* add docstrings
* black code formatting
* NOTE: odd issue where it kinda looks like i4 has to be used with FTYPE=float32, and i8 with FTYPE=float64? I didn't hunt this issue down much further, but it works when I use that so I'm sticking with it.
* NOTE: compilation time is rather slow! Possibly due to the wrapper functions. If this is an issue, we can make the wrapper functions closures (define within the actual test functions) such that importing the module should still be sufficiently fast for actual real-world usage (only testing will be slow)
